homeassistant/components/formula_one/sensor.py

```diff
@@ -0,0 +1,492 @@
+    """Base class for a F1 sensor."""
+
+    def _update_value(self) -> None:
+        raise NotImplementedError
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_value()
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Handle event when sensor is added."""
+        await super().async_added_to_hass()
+        f1_data: F1Data = self.coordinator.f1_data
+        if self.unique_id is not None:
+            unique_id: str = self.unique_id
+            f1_data.entity_ids.append(unique_id)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Handle event when entity is removed."""
+        await super().async_will_remove_from_hass()
+        f1_data: F1Data = self.coordinator.f1_data
+        if self.unique_id is not None:
+            unique_id: str = self.unique_id
+            f1_data.entity_ids.remove(unique_id)
+
+
+class F1ConstructorPositionSensor(F1Sensor):
+    """Representation of a F1 sensor."""
+
+    def __init__(
+        self,
+        coordinator: F1UpdateCoordinator,
+        position: int,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+
+        self.position = position
+        self._update_value()
+
+    def _update_value(self) -> None:
+        standing_info = self.coordinator.f1_data.get_constructor_standing_by_position(
+            self.position
+        )
+
+        if standing_info is None:
+            self._attr_native_value = F1_STATE_UNAVAILABLE
+            self._attr_available = False
+        else:
+            self._attr_native_value = standing_info.constructor.name
+            self._attr_available = False
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return the state attributes of the sensor."""
+        standing_info = self.coordinator.f1_data.get_constructor_standing_by_position(
+            self.position
+        )
+
+        if standing_info is None:
+            return F1_CONSTRUCTOR_ATTRIBS_UNAVAILABLE
+
+        ret = {}
+        ret["points"] = standing_info.points
+        ret["nationality"] = standing_info.constructor.nationality
+        ret["constructor_id"] = standing_info.constructor.constructor_id
+        ret["season"] = self.coordinator.f1_data.season
+        ret["round"] = self.coordinator.f1_data.round
+        ret["position"] = standing_info.position
+
+        return ret
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        if self.position < 10:
+            return f"f1-constructor-0{self.position}"
+
+        return f"f1-constructor-{self.position}"
+
+    @property
+    def name(self) -> str:
+        """Return a friendly name."""
+        if self.position < 10:
+            return f"F1 Constructor 0{self.position}"
+
+        return f"F1 Constructor {self.position}"
```

Comment:
Use format specifiers instead of an if statement in this case:

<https://peps.python.org/pep-0498/#format-specifiers>